File: src/Billing.Mediation/Rating/TierInput.cs
using Billing.Mediation.Context;
using Billing.Mediation.Model;

namespace Billing.Mediation.Rating;

/// <summary>Everything the rater needs for ONE tier, extracted from that tier's config snapshot by the
/// caller (so the rater stays free of the tenant-tree / config types). The leaf carries the entry
/// partner; ancestors carry the partner that owns them once the partner hierarchy is wired.</summary>
public sealed record TierInput(
    string DbName,
    int PartnerId,
    MediationContext Mediation,
    IReadOnlyList<PackageAccount> Packages,
    IReadOnlyDictionary<int, Partner> Partners);

File: java/src/main/java/com/telcobright/billing/mediation/rating/TierInput.java
package com.telcobright.billing.mediation.rating;

import com.telcobright.billing.mediation.context.MediationContext;
import com.telcobright.billing.mediation.model.PackageAccount;
import com.telcobright.billing.mediation.model.Partner;

import java.util.List;
import java.util.Map;

/**
 * Everything the rater needs for ONE tier, extracted from that tier's config snapshot by the
 * caller (so the rater stays free of the tenant-tree / config types). The leaf carries the entry
 * partner; ancestors carry the partner that owns them once the partner hierarchy is wired.
 */
public record TierInput(
        String DbName,
        int PartnerId,
        MediationContext Mediation,
        List<PackageAccount> Packages,
        Map<Integer, Partner> Partners) {
}
